force_wfmanager/ui/review/plot.py

```diff
@@ -118,14 +118,19 @@ class BasePlot(BaseDataView):
         )
     )
 
+    # --------------------
+    # Defaults and getters
+    # --------------------
+
     def _plot_updater_default(self):
-        return Timer(1000, self._check_scheduled_updates)
+        return CallbackTimer.timer(
+            interval=1, callback=self._check_scheduled_updates)
 
     def __plot_default(self):
-        self._plot = self.plot_scatter()
-        self.recenter_plot()
-        self.plot_updater.Start()
-        return self._plot
+        plot = self.plot_scatter()
+        self.plot_updater.start()
+        do_later(self.recenter_plot)
```

Comment:
The `do_later` method here calls `self.recenter_plot` after 5ms I think - is there a reason for this delay?

Reply:
Yes, `recenter_plot` acts on `self._plot` that isn't defined yet by the time this line is executed, but is just after, upon returning the default trait

<sup>(moving this comment to the relevant thread, sorry for the noise)</sup>